db/commitlog/commitlog.cc

```diff
@@ -113,6 +119,14 @@ db::commitlog::config db::commitlog::config::from_db_config(const db::config& cf
         c.commitlog_flush_threshold_in_mb = cfg.commitlog_flush_threshold_in_mb();
     }
 
+    if (c.commitlog_segment_size_in_mb < cfg.commitlog_segment_size_in_mb()) {
+        clogger.warn("Segment size {} MB too high, adjusted to {} MB", cfg.commitlog_segment_size_in_mb(), c.commitlog_segment_size_in_mb);
+    }
+    auto min_size = 2 * smp::count * c.commitlog_segment_size_in_mb;
+    if (cfg.commitlog_total_space_in_mb() >= 0 && c.commitlog_total_space_in_mb < min_size) {
```

Comment:
I don't understand this logic.

if `min_size = 2 * smp::count * c.commitlog_segment_size_in_mb` then `c.commitlog_segment_size_in_mb < min_size` always.

Reply:
Yes. But that is not the if above. It checks `commitlog_total_space_in_mb < min_size`.

Reply:
Opps, I see.

But I don't see it adjusted in this block, just warned about.